runtime_builders/publish_runtime.py

```diff
@@ -103,29 +106,34 @@ def _resolve_tag(image):
     Given a path to a tagged Docker image in GCR, replace the tag with its
     corresponding sha256 digest.
     """
-    if image is None:
-        raise Exception('Please provide image')
-
     if ':' not in image:
-        logging.info('Image does not contain tag: defaulting to "latest"')
-        base_image = image
-        target_tag = 'latest'
+        logging.error('Image must contain explicit tag!')
```

Comment:
we decided that instead of defaulting to `latest`, we should require that all images used in build steps are qualified with a tag to eliminate confusion.